include/xgboost/data.h

```diff
@@ -431,6 +447,11 @@ class DMatrix {
 
   /*! \brief page size 32 MB */
   static const size_t kPageSize = 32UL << 20UL;
+
+ protected:
```

Comment:
Can we get rid of these functions now?

Reply:
i believe this is the primary means by which the getbatches will dispatch the invocations to the appropriate sub types. hence, this may be required.

Reply:
Yes we need these methods to avoid the hacky dynamic dispatching. But they are now protected/private and not exposed to the users of DMatrix, so can be considered implementation detail.